Source/ASCollectionView.mm

```diff
@@ -317,6 +317,12 @@ - (void)reloadDataWithCompletion:(void (^)())completion
 {
   ASDisplayNodeAssertMainThread();
   
+  if (! _dataController.initialReloadDataHasBeenCalled) {
+    // If this is the first reload, forward to super immediately to prevent it from triggering more "initial" loads while our data controller is working.
+    _superIsPendingDataLoad = YES;
```

Comment:
No, it's not, but it will be right after we call `-[super reloadData]`. Turning this flag on to make sure any change set update finished before the next time super queries its data will be ignored, including the coming `reloadData` change set that we're about to fire. So this means that if `ASDataController` finishes its initial data load before the next layout pass, we'd only call `-[super reloadData]` once.